codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,443 @@
+            month = Calendar.JULY;
+        } else if (matchMonth("Aug", txt, tokenStart)) {
+            month = Calendar.AUGUST;
+        } else if (matchMonth("Sep", txt, tokenStart)) {
+            month = Calendar.SEPTEMBER;
+        } else if (matchMonth("Oct", txt, tokenStart)) {
+            month = Calendar.OCTOBER;
+        } else if (matchMonth("Nov", txt, tokenStart)) {
+            month = Calendar.NOVEMBER;
+        } else if (matchMonth("Dec", txt, tokenStart)) {
+            month = Calendar.DECEMBER;
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean tryParseYear(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len == 2) {
+            char c0 = txt.charAt(tokenStart);
+            char c1 = txt.charAt(tokenStart + 1);
+            if (isDigit(c0) && isDigit(c1)) {
+                year = getNumericalValue(c0) * 10 + getNumericalValue(c1);
+                return true;
+            }
+
+        } else if (len == 4) {
+            char c0 = txt.charAt(tokenStart);
+            char c1 = txt.charAt(tokenStart + 1);
+            char c2 = txt.charAt(tokenStart + 2);
+            char c3 = txt.charAt(tokenStart + 3);
+            if (isDigit(c0) && isDigit(c1) && isDigit(c2) && isDigit(c3)) {
+                year = getNumericalValue(c0) * 1000 +
+                        getNumericalValue(c1) * 100 +
+                        getNumericalValue(c2) * 10 +
+                        getNumericalValue(c3);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private void parseToken(CharSequence txt, int tokenStart, int tokenEnd) {
+        if (!timeFound) {
+            timeFound = tryParseTime(txt, tokenStart, tokenEnd);
+            if (timeFound) {
+                return;
+            }
+        }
+
+        if (!dayOfMonthFound) {
+            dayOfMonthFound = tryParseDayOfMonth(txt, tokenStart, tokenEnd);
+            if (dayOfMonthFound) {
+                return;
+            }
+        }
+
+        if (!monthFound) {
+            monthFound = tryParseMonth(txt, tokenStart, tokenEnd);
+            if (monthFound) {
+                return;
+            }
+        }
+
+        if (!yearFound) {
+            yearFound = tryParseYear(txt, tokenStart, tokenEnd);
+        }
+    }
+
+    private Date parse0(CharSequence txt, int start, int end) {
+        parse1(txt, start, end);
+        return normalizeAndValidate() ? computeDate() : null;
+    }
+
+    private void parse1(CharSequence txt, int start, int end) {
+        int tokenStart = -1;
+
+        for (int i = start; i < end; i++) {
+            char c = txt.charAt(i);
+
+            if (isDelim(c)) {
+                if (tokenStart != -1) {
+                    // terminate token
+                    parseToken(txt, tokenStart, i);
+                    if (allPartsFound()) {
```

Comment:
consider deriving this information from `parseToken` which already knows if everything is parsed to avoid re-evaluating all the conditionals. For example:

```java
private boolean parseToken(CharSequence txt, int tokenStart, int tokenEnd) {
// return true if you parsed all tokens, otherwise false.
}
```